packages/sdk/base/src/async.ts

```diff
@@ -139,3 +139,29 @@ export async function concurrentValuesMap<IN extends any, OUT extends any>(
     return output
   }, {})
 }
+
+export const timeout = <T extends any[], U>(
+  inFunction: InFunction<T, U>,
+  params: T,
+  timeoutMs: number,
+  timeoutRes: any,
+  logger: Logger | null = null
+) => {
+  let timer: number
+  return Promise.race([
+    inFunction(...params),
+    new Promise(() => {
```

Comment:
How does this promise ever resolve?

Reply:
setTimeout **rejects** (throws an error) to whatever you pass in as its 3rd argument when the timeout is reached. The Promise that wraps it also rejects to the same value, as does the Promise.race.